src/graph/AdminExecutor.cpp

```diff
@@ -0,0 +1,129 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "http/HttpClient.h"
+#include "graph/AdminExecutor.h"
+#include "process/ProcessUtils.h"
+#include "webservice/Common.h"
+
+#include <folly/executors/Async.h>
+#include <folly/futures/Future.h>
+
+namespace nebula {
+namespace graph {
+
+AdminExecutor::AdminExecutor(Sentence *sentence,
+        ExecutionContext *ectx) : Executor(ectx) {
+    sentence_ = static_cast<AdminSentence*>(sentence);
+    header_["add_job"] = "Job Id";
+    header_["show_jobs"] = "Job Id,Command,Status,Start Time,Stop Time";
+    header_["show_job"] = "Job Id(TaskId),Command(Dest),Status,Start Time,Stop Time";
+    header_["stop_job"] = "Job Id";
+    header_["backup_job"] = "Job num backuped";
+    header_["recover_job"] = "Job num recovered";
+}
+
+Status AdminExecutor::prepare() {
+    return Status::OK();
+}
+
+bool AdminExecutor::opNeedsSpace(const std::string& op) {
+    return op == "add_job";
+}
+
+void AdminExecutor::execute() {
+    if (opNeedsSpace(sentence_->getType())) {
+        auto status = checkIfGraphSpaceChosen();
+        if (!status.ok()) {
+            DCHECK(onError_);
+            onError_(std::move(status));
+            return;
+        }
+    }
+
+    auto *mc = ectx()->getMetaClient();
+    auto addresses = mc->getAddresses();
+    auto metaHost = network::NetworkUtils::intToIPv4(addresses[0].first);
+    auto spaceName = ectx()->rctx()->session()->spaceName();
+    auto sType = sentence_->getType();
+    std::string optionalParas("");
+    for (auto i = 0U; i != sentence_->getParas().size(); ++i) {
```

Comment:
no auto in define

Reply:
thx, change graph => meta interface from http to thrift , rewrite this executor